spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/LocallyConnected2D.scala

```diff
@@ -72,6 +73,44 @@ class LocallyConnected2D[T: ClassTag](
   require(subsample.length == 2,
     s"For LocallyConnected2D, subsample should be of length 2 but got length ${subsample.length}")
 
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toSingle().toArray
+    require(input.length == 4,
+      s"LocallyConnected2D requires 4D input, but got input dim ${input.length}")
+    val stack = if (dimOrdering == DataFormat.NCHW) (input(1), input(3), input(2))
+    else (input(3), input(2), input(1))
+    val pad = KerasUtils.getPadsFromBorderMode(borderMode)
+    val nInputPlane = stack._1
+    val inputWidth = stack._2
+    val inputHeight = stack._3
+    val nOutputPlane = nbFilter
+    val kernelW = nbCol
+    val kernelH = nbRow
+    val strideW = subsample(1)
+    val strideH = subsample(0)
+    val padW = pad._2
+    val padH = pad._1
+    val sizes =
+      if (padW == -1 && padH == -1) {
+        Utils.getSAMEOutSizeAndPadding(inputHeight, inputWidth, strideH, strideW, kernelH, kernelW)
+      } else {
+        Utils.getOutSizeAndPadding(inputHeight, inputWidth, strideH, strideW,
+          kernelH, kernelW, padH, padW, ceilMode = false)
+      }
+    val padTop = sizes(0)
+    val padBottom = sizes(1)
+    val padLeft = sizes(2)
+    val padRight = sizes(3)
+    val outputHeight = sizes(4)
+    val outputWidth = sizes(5)
```

Comment:
If we put compute output logic in the new layer, we can refer to Keras and have a much cleaner and shorter code logic for this layer and other layers as well.
Can use utils KerasUtils.computeConvOutputLength.